lib/std/os/linux/io_uring.zig

```diff
@@ -0,0 +1,877 @@
+// SPDX-License-Identifier: MIT
+// Copyright (c) 2015-2020 Zig Contributors
+// This file is part of [zig](https://ziglang.org/), which is MIT licensed.
+// The MIT license requires this copyright notice to be included in all copies
+// and substantial portions of the software.
+const std = @import("../../std.zig");
+const assert = std.debug.assert;
+const builtin = std.builtin;
+const os = std.os;
+const linux = os.linux;
+const testing = std.testing;
+
+const io_uring_params = linux.io_uring_params;
+const io_uring_sqe = linux.io_uring_sqe;
+const io_uring_cqe = linux.io_uring_cqe;
+
+pub const IO_Uring = struct {
+    fd: os.fd_t = -1,
+    sq: SubmissionQueue,
+    cq: CompletionQueue,
+    flags: u32,
+
+    /// A friendly way to setup an io_uring, with default io_uring_params.
+    /// `entries` must be a power of two between 1 and 4096, although the kernel will make the final
+    /// call on how many entries the submission and completion queues will ultimately have,
+    /// see https://github.com/torvalds/linux/blob/v5.8/fs/io_uring.c#L8027-L8050.
+    /// Matches the interface of io_uring_queue_init() in liburing.
+    pub fn init(entries: u32, flags: u32) !IO_Uring {
```

Comment:
The kernel expects it as a u32, and may increase the limit in subsequent kernels.

Reply:
Then we should allow the full `u32` range (you have an assert below); or change the signature once the kernel updates.

Reply:
I don't think we should keep changing the function signature whenever the kernel updates limits.

I do think the assertion adds value, and shouldn't be removed, otherwise the user needs to go and debug which of their args caused `EINVAL`.

Reply:
Increasing from u12 to u32 is not a breaking change, why ignore type-safety if the code does not support values bigger than that.